### ISSUE TITLE
Bar chart broken on update

### ISSUE THREAD

**Issue by vivekiitkgp** [bokeh/bokeh#3057]
Hi all,

We have been using Bokeh to generate a couple of visualizations in a project. One of the graph involves drawing a bar chart, which seems to have broken after update to 0.10. Even after putting the necessary changes, it is not clear what else needs to be done to get it working.

Previous code:
```
    p = Bar(dfs[['female', 'male']],
            stacked=True,
            xlabel="Culture",
            ylabel="Total gendered biographies",
            width=800,
            height=500,
            legend='top_left')
```

New code:
```
    p = Bar(dfs[['female', 'male']],
            stack=True,
            xlabel="Culture",
            ylabel="Total gendered biographies",
            width=800,
            height=500,
            legend='top_left')
```

Along with appropriate import changes. The new graph is, however, broken. I cannot see any drawing/axes on screen. Just a legend with `all-` label inside.

**Comment by rothnic** [comment]
@vivekiitkgp: it is hard to know without knowing the exact structure of the data. The best thing to do would be to look at the examples, which tries to show each supported use case. Looking at the configuration you posted, it appears this may just be an issue with the de-aggregated focus of the new charts, combined with a special case of stacking without any x axis labels. I have covered this case in some changes to the bar chart, which would go into the next or following release.

The charts will typically expect de-aggregated data, but can support discrete values as well. For now, I'd just avoid stacking without having some labels.

```
from bokeh.charts import Bar, show, output_file

data = {'sex': ['male', 'female'],
        'count': [54, 32]}

# once my changes are merged, you'd do
#b = Bar(data, values='count', stack='sex', ylabel='Count')

# for now, do this
b = Bar(data, values='count', label='sex', ylabel='Count')
output_file('bar_test.html')

show(b)
```

Examples file:
https://github.com/bokeh/bokeh/blob/master/examples/charts/file/bar_multi.py

**Comment by vivekiitkgp** [comment]
@rothnic Thanks for your response.

The data is categorical in nature where I have several columns listed across some categories.

**Comment by bryevdv** [comment]
@vivekiitkgp The charts in `_legacy_charts` are the older versions of charts, without any API changes. If you are going to stick with the `_legacy_charts` import for now, then nothing else should change (i.e., you also seemed to change `stacked` to `stack`, perhaps that is the problem) .

**Comment by vivekiitkgp** [comment]
@bryevdv I am currently getting it to work using `_legacy_charts` only.

**Comment by rothnic** [comment]
@vivekiitkgp: new charts will support that case with the commit referenced above, once it is merged in.

**Comment by rothnic** [comment]
Closed by the [referenced commit](https://github.com/bokeh/bokeh/commit/fa846cfd0b7f2d31af7ac3a5444f130a91fd7bbb). Will be available in the 0.11 release.
